lib/ansible/modules/cloud/vmware/vmware_drs_group_facts.py

```diff
@@ -0,0 +1,276 @@
+
+            cluster_obj = self.find_cluster_by_name(cluster_name=self.__cluster_name)
+
+            if cluster_obj is None:
+                raise Exception("Cluster '%s' not found" % self.__cluster_name)
+            else:
+                self.cluster_obj_list = [cluster_obj]
+
+    def get_result(self):
+        """
+        Docstring
+        """
+        return self.__result
+
+    def __set_result(self, result):
+        """
+        Sets result
+        Args:
+            result: drs group result list
+
+        Returns: None
+
+        """
+        self.__result = result
+
+    def __get_all_from_group(self, group_obj, host_group=False):
+        """
+        Return all VM / Host names using given group
+        Args:
+            group_obj: Group object
+            host_group: True if we want only host name from group
+
+        Returns: List of VM / Host names belonging to given group object
+
+        """
+        obj_name_list = []
+
+        if not all([group_obj]):
+            return obj_name_list
+
+        if not host_group and isinstance(group_obj, vim.cluster.VmGroup):
+            obj_name_list = [vm.name for vm in group_obj.vm]
+        elif host_group and isinstance(group_obj, vim.cluster.HostGroup):
+            obj_name_list = [host.name for host in group_obj.host]
+
+        return obj_name_list
+
+    def __normalize_group_data(self, group_obj):
+        """
+        Return human readable group spec
+        Args:
+            group_obj: Group object
+
+        Returns: Dictionary with DRS groups
+
+        """
+        if not all([group_obj]):
+            return {}
+
+        # Check if group is a host group
+        if hasattr(group_obj, 'host'):
+            return dict(
+                group_name=group_obj.name,
+                hosts=self.__get_all_from_group(group_obj=group_obj, host_group=True),
+                type="host"
+            )
+        else:
+            return dict(
+                group_name=group_obj.name,
+                vms=self.__get_all_from_group(group_obj=group_obj),
+                type="vm"
+            )
+
+    def gather_facts(self):
+        """
+        Gather DRS group facts about given cluster
+        Returns: Dictionary of clusters with DRS groups
+
+        """
+        cluster_group_facts = dict()
+
+        for cluster_obj in self.cluster_obj_list:
+
+            cluster_group_facts[cluster_obj.name] = []
+
+            for drs_group in cluster_obj.configurationEx.group:
+                cluster_group_facts[cluster_obj.name].append(self.__normalize_group_data(drs_group))
+
+        self.__set_result(cluster_group_facts)
+
+
+def main():
+
+    argument_spec = vmware_argument_spec()
+
+    argument_spec.update(
+        datacenter=dict(type='str', required=True, aliases=['datacenter_name']),
+        cluster_name=dict(type='str', required=False),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
```

Comment:
What happens if both `cluster_name` and `datacenter` is set? Should there be a `required_one_of` added here, and a `mutually_exclusive`?
```suggestion
        supports_check_mode=True,
        required_one_of=[['cluster_name', 'datacenter']],
        mutually_exclusive=[['cluster_name', 'datacenter']],
```

Reply:
You are absolutely right. No need for `cluster_name` if `datacenter` is set.